### PR TITLE
Remove unnecessary async in public API

### PR DESCRIPTION
**Pull request by alexlapa**
> mind if i create a PR that makes all callback setters (e.g. RTCPeerConnection::on_ice_candidate()) non async? Currently callbacks are stored behind tokio::Mutex'es, i believe that ArcSwap can be used here, so we can use atomic load/store. It wont affect performance in any way, just some quality of life improvements.

> All good with me. I'd like to get rid of  tokio::Mutex, replacing it with sync variants, where that's possible

So i've refactored all public callback setters here. 

Regarding removing `tokio::Mutex`es that you've mentioned, well, i've replaced some of those, but i don't want this PR to become even bigger, and it would be nice if you provide some feedback first. I've used both `ArcSwap` and `std::Mutex`, i guess its ok to use blocking mutex in simple cases, when you don't need to hold it, e.g. copy, store, swap, take one-liners. I will prepare follow-up PRs on this issue if this PR looks good to you.

And one more thing regarding the callbacks, maybe it makes sense to make them just dumb `Box<dyn Fn(_)>`? So `Box<dyn (FnMut(ConnectionState) -> Pin<Box<dyn Future<Output = ()> + Send + 'static>>) + Send + Sync>` => `Box<dyn (Fn(ConnectionState)) + Send + Sync>`.  Yeah, that might hurt users in some rare cases, but it would make everything simpler.

ack @k0nserv

### CODE REVIEW

File: ice/src/agent/agent_internal.rs [hunk 1]
```diff
@@ -1085,14 +1079,14 @@ impl AgentInternal {
                 tokio::select! {
                     opt_state = chan_state_rx.recv() => {
                         if let Some(s) = opt_state {
-                            let mut on_connection_state_change_hdlr = ai.on_connection_state_change_hdlr.lock().await;
-                            if let Some(f) = &mut *on_connection_state_change_hdlr{
+                            if let Some(hndlr) = &*ai.on_connection_state_change_hdlr.load() {
```
Comment (melekes) [inline ice/src/agent/agent_internal.rs:1082]:
```suggestion
                            if let Some(handler) = &*ai.on_connection_state_change_hdlr.load() {
```

let's maybe give this a more human readable name

File: ice/src/agent/agent_selector.rs [hunk 2]
```diff
@@ -61,33 +61,33 @@ trait ControlledSelector {
 }
 
 impl AgentInternal {
-    async fn is_nominatable(&self, c: &Arc<dyn Candidate + Send + Sync>) -> bool {
-        let start_time = self.start_time.lock().await;
+    fn is_nominatable(&self, c: &Arc<dyn Candidate + Send + Sync>) -> bool {
+        let start_time = { *self.start_time.lock().unwrap() };
```
Comment (melekes) [inline ice/src/agent/agent_selector.rs:65]: could you add a comment stating why we'll never panic on unwrap here? also in general `expect("..")` is better in production code (unwrap is ok for tests)

File: webrtc/src/data_channel/mod.rs [hunk 3]
```diff
@@ -67,10 +72,10 @@ pub struct RTCDataChannel {
     // is created, the binaryType attribute MUST be initialized to the string
     // "blob". This attribute controls how binary data is exposed to scripts.
     // binaryType                 string
-    pub(crate) on_message_handler: Arc<Mutex<Option<OnMessageHdlrFn>>>,
-    pub(crate) on_open_handler: Arc<Mutex<Option<OnOpenHdlrFn>>>,
-    pub(crate) on_close_handler: Arc<Mutex<Option<OnCloseHdlrFn>>>,
-    pub(crate) on_error_handler: Arc<Mutex<Option<OnErrorHdlrFn>>>,
+    pub(crate) on_message_handler: Arc<ArcSwapOption<Mutex<OnMessageHdlrFn>>>,
```
Comment (melekes) [inline webrtc/src/data_channel/mod.rs:75]: why do we need Arc inside another Arc here?

File: webrtc/src/data_channel/mod.rs [hunk 4]
```diff
@@ -67,10 +72,10 @@ pub struct RTCDataChannel {
...
+    pub(crate) on_open_handler: StdMutex<Option<OnOpenHdlrFn>>,
```
Comment (melekes) [inline webrtc/src/data_channel/mod.rs:76]: and why there's no longer an Arc here? What if the user clones `DataChannel`?

File: webrtc/src/data_channel/mod.rs (hunk 3 above)
Comment (alexlapa) [reply webrtc/src/data_channel/mod.rs:75]: `ArcSwapOption` is basically an `Option<Arc<T>>` not `Arc<Option<Arc<T>>>`, it has no `Clone` implementation, and must be wrapped into another `Arc` if it needs to be shared. We can only clone the inner `Arc`, but it won't be changed if we call `ArcSwapOption::store` after.

File: webrtc/src/data_channel/mod.rs (hunk 4 above)
Comment (alexlapa) [reply webrtc/src/data_channel/mod.rs:76]: You mean `Arc<Mutex<Option<OnOpenHdlrFn>>>` => `StdMutex<Option<OnOpenHdlrFn>>`? Well i [refactored](https://www.diffchecker.com/E41ixG2e) a `RTCDataChannel::do_open` a bit, it used to clone this `Arc` into `tokio::spawn`, and now it takes the inner `OnMessageHdlrFn` from `Option` and moves it to that `tokio::spawn`.  I believe its ok, let me know if you want this reverted.

File: ice/src/agent/agent_internal.rs (hunk 1 above)
Comment (alexlapa) [reply ice/src/agent/agent_internal.rs]: Done [87fca0e](https://github.com/webrtc-rs/webrtc/pull/338/commits/87fca0e4031fde7b8a9cecf37e1878c2dc2096f1)

File: ice/src/agent/agent_selector.rs (hunk 2 above)
Comment (alexlapa) [reply ice/src/agent/agent_selector.rs]: Done [87fca0e](https://github.com/webrtc-rs/webrtc/pull/338/commits/87fca0e4031fde7b8a9cecf37e1878c2dc2096f1)

**Comment by lolgesten** [comment]
There's like way too much to read in detail here, but I'm so here for it. Thanks!

**Comment by algesten** [comment]
I love this. Thank you so much for taking the time to do it!

(ignore my approve from work github user above)

**Review by k0nserv — COMMENTED** [review]
Awesome job 👍🏼 

I had a few notes. The point about wrapping the sync primitives is not something we have to do, but if you feel up for it it'd be great.

Overall this is excellent

File: webrtc/src/data_channel/mod.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (melekes) [reply webrtc/src/data_channel/mod.rs:75]: https://github.com/webrtc-rs/webrtc/pull/338#discussion_r1013784116

File: webrtc/src/data_channel/mod.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (melekes) [reply webrtc/src/data_channel/mod.rs:76]: https://github.com/webrtc-rs/webrtc/pull/338#discussion_r1013799574

File: examples/examples/broadcast/broadcast.rs [hunk 5]
```diff
@@ -127,72 +127,68 @@ async fn main() -> Result<()> {
...
-            },
-        ))
-        .await;
+            Box::pin(async {})
+        },
+    ));
 
     // Set the handler for Peer connection state
     // This will notify you when the peer has connected/disconnected
-    peer_connection
```
[the remark this answers is not in the document]
Comment (alexlapa) [reply examples/examples/broadcast/broadcast.rs:190]: Well i removed `.await` there, default rustfmt's `max_width` is 100 and this line is 96 now so its ok. You can override that with `.rustfmt.toml` if you want. But the only reason that may be worth doing is `imports_granularity = "Crate"`, absolutely love that.

[the remark this answers is not in the document]
Comment (k0nserv) [reply examples/examples/broadcast/broadcast.rs:190]: Of course, my bad

**Comment by alexlapa** [comment]
@k0nserv , fixed

File: util/src/conn/conn_disconnected_packet.rs [hunk 6]
```diff
@@ -30,11 +30,7 @@ impl Conn for DisconnectedPacketConn {
     async fn recv(&self, buf: &mut [u8]) -> Result<usize> {
         let (n, addr) = self.pconn.recv_from(buf).await?;
         {
-            // Won't panic since we only do set/get one-liners.
-            *self
-                .raddr
-                .lock()
-                .expect("DisconnectedPacketConn::raddr is poisoned") = addr;
+            *self.raddr.write() = addr;
         }
```
Comment (melekes) [inline util/src/conn/conn_disconnected_packet.rs:34]:
```suggestion
```

File: util/src/vnet/conn.rs [hunk 7]
```diff
@@ -86,10 +82,7 @@ impl Conn for UdpConn {
     /// the n > 0 bytes returned before considering the error err.
     async fn recv_from(&self, buf: &mut [u8]) -> Result<(usize, SocketAddr)> {
         let mut read_ch = self.read_ch_rx.lock().await;
-        let rem_addr = {
-            // Won't panic since we only do set/get one-liners.
-            *self.rem_addr.lock().expect("UdpConn::rem_addr is poisoned")
-        };
+        let rem_addr = { *self.rem_addr.read() };
```
Comment (melekes) [inline util/src/vnet/conn.rs:85]:
```suggestion
        let rem_addr = *self.rem_addr.read();
```

File: util/src/vnet/conn.rs [hunk 8]
```diff
@@ -109,10 +102,7 @@ impl Conn for UdpConn {
     }
 
     async fn send(&self, buf: &[u8]) -> Result<usize> {
-        let rem_addr = {
-            // Won't panic since we only do set/get one-liners.
-            *self.rem_addr.lock().expect("UdpConn::rem_addr is poisoned")
-        };
+        let rem_addr = { *self.rem_addr.read() };
```
Comment (melekes) [inline util/src/vnet/conn.rs:105]:
```suggestion
        let rem_addr = *self.rem_addr.read();
```

File: webrtc/src/data_channel/mod.rs [hunk 9]
```diff
@@ -207,12 +208,7 @@ impl RTCDataChannel {
     /// the underlying data transport has been established (or re-established).
     pub fn on_open(&self, f: OnOpenHdlrFn) {
         {
```
Comment (melekes) [inline webrtc/src/data_channel/mod.rs:210]:
```suggestion
```

File: webrtc/src/data_channel/mod.rs [hunk 10]
```diff
@@ -207,12 +208,7 @@ impl RTCDataChannel {
...
-            // Won't panic since we only do set/take one-liners.
-            let _ = self
-                .on_open_handler
-                .lock()
-                .expect("RTCDataChannel::on_open_handler is poisoned")
-                .replace(f);
+            let _ = self.on_open_handler.lock().replace(f);
         }
```
Comment (melekes) [inline webrtc/src/data_channel/mod.rs:212]:
```suggestion
```

File: webrtc/src/data_channel/mod.rs [hunk 11]
```diff
@@ -221,13 +217,7 @@ impl RTCDataChannel {
...
 
     fn do_open(&self) {
-        let on_open_handler = {
-            // Won't panic since we only do set/take one-liners.
-            self.on_open_handler
-                .lock()
-                .expect("RTCDataChannel::on_open_handler is poisoned")
-                .take()
-        };
+        let on_open_handler = { self.on_open_handler.lock().take() };
```
Comment (melekes) [inline webrtc/src/data_channel/mod.rs:220]:
```suggestion
        let on_open_handler = self.on_open_handler.lock().take();
```

File: turn/src/allocation/mod.rs [hunk 12]
```diff
@@ -288,26 +284,13 @@ impl Allocation {
...
     // Refresh updates the allocations lifetime
     pub async fn refresh(&self, lifetime: Duration) {
-        let reset_tx = {
-            // Won't panic since we only do set/take/clone one-liners.
-            self.reset_tx
-                .lock()
-                .expect("Allocation::reset_tx is poisoned")
-                .clone()
-        };
+        let reset_tx = { self.reset_tx.lock().clone() };
```
[the remark this answers is not in the document]
Comment (alexlapa) [reply turn/src/allocation/mod.rs:293]: Sure, that makes sense, but there is also this https://github.com/webrtc-rs/webrtc/pull/288#discussion_r992094353 ¯\\_(ツ)_/¯

File: util/CHANGELOG.md [hunk 13]
```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+#### Breaking changes
```
[the remark this answers is not in the document]
Comment (alexlapa) [reply util/CHANGELOG.md]: https://github.com/webrtc-rs/webrtc/pull/338/commits/9321c1c69fc843fd5efacc92b4119d8dd3897f4f

File: turn/src/allocation/mod.rs (hunk 12 above)
[the remark this answers is not in the document]
Comment (alexlapa) [reply turn/src/allocation/mod.rs]: https://github.com/webrtc-rs/webrtc/pull/338/commits/5a28ece0fc03a3abfd23a7f9c6e05cc31cad827e

**Comment by alexlapa** [comment]
@melekes , fixed

File: turn/src/allocation/mod.rs (hunk 12 above)
[the remark this answers is not in the document]
Comment (melekes) [reply turn/src/allocation/mod.rs]: > Sure, that makes sense, but there is also this [#288 (comment)](https://github.com/webrtc-rs/webrtc/pull/288#discussion_r992094353) ¯\_(ツ)_/¯

what @k0nserv was referring to I guess is smth like

```rust
{ 
  let reset_tx = self.reset_tx.lock().clone();
  // do something with reset
} // lock is dropped
```

here you just wrapping expression in unnecessary braces.

**Review by k0nserv — APPROVED** [review]
LGTM, I'm just going to test this out in a real world application and see that it doesn't obviously break anything

**Comment by k0nserv** [comment]
When several peers join in quick succession I end up stuck with this warning repeating endlessly


```
webrtc_ice::agent::agent_internal [controlled]: pingAllCandidates called with no candidate pairs. Connection is not possible yet. 
```

**Comment by k0nserv** [comment]
I did some more digging into the ICE problem. It happens when webrtc-rs is the controlled ICE agent. There's a STUN binding request form the controlling agent, but it's never read by webrtc-rs. I suspect there's a deadlock or something like that somehwere

File: webrtc/src/ice_transport/ice_gatherer.rs [hunk 14]
```diff
@@ -84,8 +85,7 @@ impl RTCIceGatherer {
         // ensure we do not do anything expensive other than the actual agent
         // creation in this function.
 
-        let mut agent = self.agent.lock().await;
```
Comment (k0nserv) [inline webrtc/src/ice_transport/ice_gatherer.rs:87]: This is a regression, we create duplicated agents again. See the comment above this

File: turn/src/allocation/mod.rs [hunk 15]
```diff
@@ -282,15 +282,13 @@ impl Allocation {
...
-        reset_tx.take();
-        expired
+        let reset_tx = self.reset_tx.lock().take();
+        reset_tx.is_none() || self.timer_expired.load(Ordering::SeqCst)
     }
 
     // Refresh updates the allocations lifetime
     pub async fn refresh(&self, lifetime: Duration) {
-        let reset_tx = self.reset_tx.lock().unwrap().clone();
+        let reset_tx = self.reset_tx.lock().clone();
```
[the remark this answers is not in the document]
Comment (k0nserv) [reply turn/src/allocation/mod.rs:291]: It doesn't, I [double checked](https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=1b9523ceb7276f5d04adb206e0e3391c)

[the remark this answers is not in the document]
Comment (alexlapa) [reply turn/src/allocation/mod.rs:291]: Y, `MutexGuard` does not have a `Clone` implementation, so it is:
```rust
let reset_tx: Option<mpsc::Sender<Duration>> = self.reset_tx.lock().clone();
```

**Comment by k0nserv** [comment]
Been using this to play around with detecting deadlocks, didn't find anything though. 

```diff
diff --git a/util/src/sync/mod.rs b/util/src/sync/mod.rs
index 6a101299..d5eb835d 100644
--- a/util/src/sync/mod.rs
+++ b/util/src/sync/mod.rs
@@ -1,4 +1,150 @@
-use std::{ops, sync};
+use std::backtrace::Backtrace;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+use std::{fmt, ops, sync, thread};
+
+const MAX_AGE: Duration = Duration::from_millis(5000);
+const CHECK_INTERVAL: Duration = Duration::from_millis(20);
+static DEADLOCK_THREAD_STARTED: AtomicBool = AtomicBool::new(false);
+lazy_static! {
+    static ref TRACKED_LOCKS: Arc<sync::RwLock<Vec<TrackedLock>>> =
+        Arc::new(sync::RwLock::new(Vec::new()));
+}
+
+struct TrackedLock {
+    backtrace: Backtrace,
+    tracker: AliveTracker,
+    typ: Type,
+    reported: AtomicBool,
+}
+
+impl TrackedLock {
+    fn mark_reported(&self) -> bool {
+        self.reported
+            .compare_exchange(false, true, Ordering::Acquire, Ordering::Acquire)
+            .is_ok()
+    }
+}
+
+#[derive(Debug)]
+enum Type {
+    MutexLock,
+    RwReadLock,
+    RwWriteLock,
+}
+
+impl fmt::Display for Type {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Type::MutexLock => write!(f, "MutexLock"),
+            Type::RwReadLock => write!(f, "RwReadLock"),
+            Type::RwWriteLock => write!(f, "RwWriteLock"),
+        }
+    }
+}
+
+struct Alive {
+    live: Arc<AtomicBool>,
+    born_at: Instant,
+}
+
+impl Alive {
+    fn new() -> Self {
+        Self {
+            live: Arc::new(AtomicBool::new(true)),
+            born_at: Instant::now(),
+        }
+    }
+
+    fn track(&self) -> AliveTracker {
+        AliveTracker {
+            live: self.live.clone(),
+            born_at: self.born_at.clone(),
+        }
+    }
+
+    fn mark_dead(&self) {
+        self.live.store(false, Ordering::Release);
+    }
+}
+
+impl Drop for Alive {
+    fn drop(&mut self) {
+        self.mark_dead();
+    }
+}
+
+struct AliveTracker {
+    live: Arc<AtomicBool>,
+    born_at: Instant,
+}
+
+impl AliveTracker {
+    fn is_dead(&self) -> bool {
+        !self.live.load(Ordering::Acquire)
+    }
+
+    fn age(&self) -> Option<Duration> {
+        if self.is_dead() {
+            None
+        } else {
+            Some(Instant::now() - self.born_at)
+        }
+    }
+}
+
+fn start_deadlock_thread() {
+    if DEADLOCK_THREAD_STARTED
+        .compare_exchange(false, true, Ordering::Acquire, Ordering::Acquire)
+        .is_err()
+    {
+        return;
+    }
+
+    thread::spawn(|| loop {
+        let lock = TRACKED_LOCKS.read().unwrap();
+        let mut needs_cleanup = false;
+
+        for tracked in lock.iter() {
+            let age = tracked.tracker.age();
+            match age {
+                Some(age) if age > MAX_AGE => {
+                    if tracked.mark_reported() {
+                        println!(
+                            "Potential deadlock on {}. Backtrace:\n{}Held for: {}s",
+                            tracked.typ,
+                            tracked.backtrace,
+                            age.as_secs_f64()
+                        );
+                    }
+                }
+                None => {
+                    // Lock is dead
+                    needs_cleanup = true;
+                }
+                _ => {}
+            }
+        }
+        drop(lock);
+
+        if needs_cleanup {
+            let mut write = TRACKED_LOCKS.write().unwrap();
+            write.retain(|tracked| !tracked.tracker.is_dead());
+        }
+    });
+}
+
+fn track(backtrace: Backtrace, alive: &Alive, typ: Type) {
+    start_deadlock_thread();
+    let mut write = TRACKED_LOCKS.write().unwrap();
+    write.push(TrackedLock {
+        backtrace,
+        tracker: alive.track(),
+        typ,
+        reported: AtomicBool::new(false),
+    });
+}
 
 /// A synchronous mutual exclusion primitive useful for protecting shared data.
 #[derive(Default, Debug)]
@@ -13,8 +159,10 @@ impl<T> Mutex<T> {
     /// Acquires a mutex, blocking the current thread until it is able to do so.
     pub fn lock(&self) -> MutexGuard<'_, T> {
         let guard = self.0.lock().unwrap();
+        let alive = Alive::new();
+        track(Backtrace::capture(), &alive, Type::MutexLock);
 
-        MutexGuard(guard)
+        MutexGuard(guard, alive)
     }
 
     /// Consumes this mutex, returning the underlying data.
@@ -25,7 +173,7 @@ impl<T> Mutex<T> {
 
 /// An RAII implementation of a "scoped lock" of a mutex. When this structure is
 /// dropped (falls out of scope), the lock will be unlocked.
-pub struct MutexGuard<'a, T>(sync::MutexGuard<'a, T>);
+pub struct MutexGuard<'a, T>(sync::MutexGuard<'a, T>, Alive);
 
 impl<'a, T> ops::Deref for MutexGuard<'a, T> {
     type Target = T;
@@ -55,22 +203,26 @@ impl<T> RwLock<T> {
     /// until it can be acquired.
     pub fn read(&self) -> RwLockReadGuard<'_, T> {
         let guard = self.0.read().unwrap();
+        let alive = Alive::new();
+        track(Backtrace::capture(), &alive, Type::MutexLock);
 
-        RwLockReadGuard(guard)
+        RwLockReadGuard(guard, alive)
     }
 
     /// Locks this rwlock with exclusive write access, blocking the current
     /// thread until it can be acquired.
     pub fn write(&self) -> RwLockWriteGuard<'_, T> {
         let guard = self.0.write().unwrap();
+        let alive = Alive::new();
+        track(Backtrace::capture(), &alive, Type::MutexLock);
 
-        RwLockWriteGuard(guard)
+        RwLockWriteGuard(guard, alive)
     }
 }
 
 /// RAII structure used to release the shared read access of a lock when
 /// dropped.
-pub struct RwLockReadGuard<'a, T>(sync::RwLockReadGuard<'a, T>);
+pub struct RwLockReadGuard<'a, T>(sync::RwLockReadGuard<'a, T>, Alive);
 
 impl<'a, T> ops::Deref for RwLockReadGuard<'a, T> {
     type Target = T;
@@ -82,7 +234,7 @@ impl<'a, T> ops::Deref for RwLockReadGuard<'a, T> {
 
 /// RAII structure used to release the shared read access of a lock when
 /// dropped.
-pub struct RwLockWriteGuard<'a, T>(sync::RwLockWriteGuard<'a, T>);
+pub struct RwLockWriteGuard<'a, T>(sync::RwLockWriteGuard<'a, T>, Alive);
 
 impl<'a, T> ops::Deref for RwLockWriteGuard<'a, T> {
     type Target = T;
```

File: webrtc/src/ice_transport/ice_gatherer.rs (hunk 14 above)
Comment (alexlapa) [reply webrtc/src/ice_transport/ice_gatherer.rs:87]: Fixed

File: sctp/Cargo.toml [hunk 16]
```diff
@@ -15,6 +15,7 @@ rust-version = "1.60.0"
 [dependencies]
 util = { version = "0.6.0", path = "../util", package = "webrtc-util", default-features = false, features = ["conn"] }
 
+arc-swap = "1.5.1"
```
Comment (melekes) [inline sctp/Cargo.toml:18]:
```suggestion
arc-swap = "1.5"
```

File: webrtc/Cargo.toml [hunk 17]
```diff
@@ -29,6 +29,7 @@ stun = { version = "0.4.3", path = "../stun" }
 turn = { version = "0.6.0", path = "../turn" }
 util = { version = "0.6.0", path = "../util", package = "webrtc-util" }
 
+arc-swap = "1.5.1"
```
Comment (melekes) [inline webrtc/Cargo.toml:32]:
```suggestion
arc-swap = "1.5"
```

File: ice/Cargo.toml [hunk 18]
```diff
@@ -18,6 +18,7 @@ turn = { version = "0.6.0", path = "../turn" }
 stun = { version = "0.4.3", path = "../stun" }
 mdns = { version = "0.5.0", path = "../mdns", package = "webrtc-mdns" }
 
+arc-swap = "1.5.1"
```
Comment (melekes) [inline ice/Cargo.toml:21]:
```suggestion
arc-swap = "1.5"
```

**Review by melekes — APPROVED** [review]

**Comment by alexlapa** [comment]
Done b31ec32de2e64232595aa9238cea6984cf9193ff.

I believe that CI failure is irrelevant, since [latest master](https://github.com/webrtc-rs/webrtc/actions/runs/3442468054/jobs/5743076538) also fails.

**Comment by melekes** [comment]
```
test data_channel::data_channel_test::test_data_channel_buffered_amount has been running for over 60 seconds
```

restarted tests to see if ^ persists.

**Comment by melekes** [comment]
> ```
> test data_channel::data_channel_test::test_data_channel_buffered_amount has been running for over 60 seconds
> ```
> 
> restarted tests to see if ^ persists.

looks like a non-deterministic failure.

**Pull request merged**